File: Week2/Python/ArrayProblem/LargestElement.py
"""
Given an array, arr. The task is to find the largest element in it.

https://www.geeksforgeeks.org/problems/largest-element-in-array4009/1
"""

from typing import List


def largest(arr : List[int]) -> int:
    """ T(c) -> O(n) , S(c) -> O(1) """
    
    maxNum = arr[0]

    for i in range(1, len(arr)):
        if arr[i] > maxNum:
            maxNum = arr[i]
    
    return maxNum


arr = [1, 8, 7, 56, 90, 76]
print(largest(arr))

File: Week2/Java/ArrayProblem/LargestElement.java
public class LargestElement{
/*
 * Given an array A[] of size n. The task is to find the largest element in it.
 * https://www.geeksforgeeks.org/problems/largest-element-in-array4009/1
 * T(c) -> O(n), S(c) -> O(1)
 */
    static int largest(int arr[], int n)
    {
        int max = Integer.MIN_VALUE;
        for(int i=0;i<n;i++)
        {
            if(arr[i] > max)
            {
                max = arr[i];
            }
        }
        return max;
    }

    public static void main(String[] args) {
        int arr[] = {10,5,4,13};
        System.out.println(largest(arr, arr.length));
    }   
}